spec/lib/jobs/ucs-service-profile-discovery-spec.js

```diff
@@ -0,0 +1,146 @@
+// Copyright 2017, EMC, Inc.
+
+'use strict';
+describe('Ucs Discovery Job', function () {
+    var uuid = require('node-uuid'),
+        sandbox = sinon.sandbox.create(),
+        graphId = uuid.v4(),
+        ucsJob,
+        ucsTool,
+        rackmountData,
+        rootData,
+        serverData,
+        waterline = {},
+        Error;
+
+
+    var obm = {
+        config: {
+            uri: "http://10.240.19.220:7080/sys",
+            host: "10.240.19.220",
+            root: "/sys",
+            port: "7080",
+            protocol: "http",
+            username: "ucspe",
+            password: "ucspe",
+            ucs: "10.239.85.153",
+            verifySSL: false
+        },
+        service: "ucs-obm-service"
+    };
+
+    var node = {
+        id: 'abc',
+        type: 'compute',
+        name: 'ls-testServiceProfile',
+        associatedServer: 'sys/test-chassis/test-server',
+        identifiers: [],
+        relations: [],
+        obm: [obm]
+    };
+    
+    before(function() { 
+        helper.setupInjector([
+            helper.require('/lib/jobs/base-job.js'),
+            helper.require('/lib/jobs/ucs-service-profile-discovery.js'),
+            helper.require('/lib/utils/job-utils/ucs-tool.js'),
+            helper.di.simpleWrapper(waterline,'Services.Waterline')
+        ]);
+        waterline.nodes = {
+            create: sandbox.stub().resolves(node),
+            needOne: sandbox.stub().resolves(node),
+            updateOne: sandbox.stub().resolves(node),
+            //needOneById: sandbox.stub().resolves(node)
+        };
+        waterline.obms = {
+            upsertByNode: sandbox.stub().resolves(obm)
+        };
+        Error = helper.injector.get('Errors');
+    });
+    
+    afterEach(function() {
+        sandbox.restore();
+    });
+    
+    beforeEach(function() {
+        var Job = helper.injector.get('Job.Ucs.Service.Profile.Discovery');
+        ucsJob = new Job({
+            uri:'https://1.1.1.1:7080/sys',
+            username:'user',
+            password:'pass',
+            ucs:"1.1.1.1"
+        }, {}, graphId);
```

Comment:
this is `taskId`, not `graphId`, though this doesn't break the unit-test, but don't bring confusion since unit-test is a kind of doc for the code.